pkg/apis/core/validation/validation.go

```diff
@@ -1507,12 +1539,10 @@ func validateStorageOSPersistentVolumeSource(storageos *core.StorageOSPersistent
 		allErrs = append(allErrs, ValidateDNS1123Label(storageos.VolumeNamespace, fldPath.Child("volumeNamespace"))...)
 	}
 	if storageos.SecretRef != nil {
-		if len(storageos.SecretRef.Name) == 0 {
-			allErrs = append(allErrs, field.Required(fldPath.Child("secretRef", "name"), ""))
-		}
-		if len(storageos.SecretRef.Namespace) == 0 {
-			allErrs = append(allErrs, field.Required(fldPath.Child("secretRef", "namespace"), ""))
-		}
+		allErrs = append(allErrs, ValidateSecretReference(&core.SecretReference{
```

Comment:
I would prefer if there was`ValidateLocalSecretReference()` instead of calling ValidateSecretReference with a fake object.

Reply:
I understand the concern but I don't quite get the solution
At the risk of sounding lame I have to ask 
Can you please share some reference?

Reply:
Sorry, `ValidateLocalSecretReference` was not a good name. `storageos.SecretRef` is a pointer to `ObjectReference`. I am proposing:

`func ValidateSecretObjectReference(ref *core.ObjectReference) []error` that checks if `ObjectReference` is really a Secret + has valid name + namespace.

Reply:
This makes sense
I'll update the PR

Reply:
@jsafrane 
I have added ValidateSecretObjectReference method that follows the criteria you mentioned above